endpoints/setuid.go

```diff
@@ -44,64 +51,64 @@ func NewSetUIDEndpoint(cfg config.HostCookie, perms gdpr.Permissions, pbsanalyti
 		}
 
 		query := r.URL.Query()
-		bidder := query.Get("bidder")
-		if shouldReturn, status, body := preventSyncsGDPR(query.Get("gdpr"), query.Get("gdpr_consent"), perms); shouldReturn {
-			w.WriteHeader(status)
-			w.Write([]byte(body))
-			metrics.RecordUserIDSet(pbsmetrics.UserLabels{
-				Action: pbsmetrics.RequestActionGDPR,
-				Bidder: openrtb_ext.BidderName(bidder),
-			})
-			so.Status = status
-			return
-		}
 
-		if err := validateBidder(bidder); err != nil {
+		familyName, err := getFamilyName(query, validFamilyNameMap)
+		if err != nil {
 			w.WriteHeader(http.StatusBadRequest)
 			w.Write([]byte(err.Error()))
 			metrics.RecordUserIDSet(pbsmetrics.UserLabels{
 				Action: pbsmetrics.RequestActionErr,
-				Bidder: openrtb_ext.BidderName(bidder),
 			})
 			so.Status = http.StatusBadRequest
 			return
 		}
-		so.Bidder = bidder
+		so.Bidder = familyName
+
+		if shouldReturn, status, body := preventSyncsGDPR(query.Get("gdpr"), query.Get("gdpr_consent"), perms); shouldReturn {
+			w.WriteHeader(status)
+			w.Write([]byte(body))
+			metrics.RecordUserIDSet(pbsmetrics.UserLabels{
+				Action: pbsmetrics.RequestActionGDPR,
+				Bidder: openrtb_ext.BidderName(familyName),
+			})
+			so.Status = status
+			return
+		}
 
 		uid := query.Get("uid")
 		so.UID = uid
 
-		var err error
 		if uid == "" {
-			pc.Unsync(bidder)
+			pc.Unsync(familyName)
 		} else {
-			err = pc.TrySync(bidder, uid)
-		}
-
-		if err == nil {
-			labels := pbsmetrics.UserLabels{
-				Action: pbsmetrics.RequestActionSet,
-				Bidder: openrtb_ext.BidderName(bidder),
+			if err = pc.TrySync(familyName, uid); err != nil {
```

Comment:
Should this be `err == nil`? ... it seems you are coding for success here.

Reply:
Correct. I'll fix and add a test.